package/base-files/files/lib/upgrade/common.sh

```diff
@@ -207,17 +218,35 @@ get_partitions() { # <device> <filename>
 		rm -f "/tmp/partmap.$filename"
 
 		local part
-		for part in 1 2 3 4; do
-			set -- $(hexdump -v -n 12 -s "$((0x1B2 + $part * 16))" -e '3/4 "0x%08X "' "$disk")
-
-			local type="$(( $(hex_le32_to_cpu $1) % 256))"
-			local lba="$(( $(hex_le32_to_cpu $2) ))"
-			local num="$(( $(hex_le32_to_cpu $3) ))"
-
-			[ $type -gt 0 ] || continue
-
-			printf "%2d %5d %7d\n" $part $lba $num >> "/tmp/partmap.$filename"
-		done
+		local magic=$(dd if="$disk" bs=8 count=1 skip=64 2>/dev/null)
```

Comment:
Same with this magic, used also many times

Reply:
Maybe I should make it into a function。

Reply:
yes, usually if you use same code (copy&paste it) more then one time, it's good idea to make it a common function and reuse it